### ISSUE TITLE
Crash in compiler when using `extends null`, `super()`, and re-opening class with `interface`

### ISSUE THREAD

**Issue by rbuckton** [microsoft/TypeScript#55499]
### 🔎 Search Terms

extends null, crash

### 🕗 Version & Regression Information

- This is a crash
- This is new since 5.2

### ⏯ Playground Link

https://www.typescriptlang.org/play?lib=true&ts=5.2.1-rc#code/JYOwLgpgTgZghgYwgAgEJwM4oN4CgC+uuCANphsgMLIQAekIAJhSAK4knJ7I-IID2IDGCisEYflAAUASi65ei5BlYAHaLIDcC3oUKhIsRCmp0GzNJhz4gA

### 💻 Code

```ts
interface Base {
}

class C extends null {
    constructor() {
        super(); // (1)
    }
}
interface C extends Base {} // (2)
```

### 🙁 Actual behavior

Crash when both `class C extends null` and `interface C extends Base` are used. If you comment out (1), then there is no crash and no error (expected). If you comment out (2), there is no crash and a diagnostic is reported at (1) (expected). The combination of both results in an error.

### 🙂 Expected behavior

The diagnostic message `A constructor cannot contain a 'super' call when its class extends 'null'.` should be printed, just as it does when removing the `interface C extends Base {}` line.

### Additional information about the issue

The crash has the following stack trace:

```
Error: Debug Failure. False expression: Revert #54442 and add a testcase with whatever triggered this
    at resolveCall (...\tsc.js:70670:11)
    at resolveCallExpression (...\tsc.js:71041:18)
    at resolveSignature (...\tsc.js:71486:16)
    at getResolvedSignature (...\tsc.js:71506:18)
    at checkCallExpression (...\tsc.js:71618:23)
    at checkExpressionWorker (...\tsc.js:74800:16)
    at checkExpression (...\tsc.js:74706:32)
    at checkExpressionStatement (...\tsc.js:77336:5)
    at checkSourceElementWorker (...\tsc.js:80280:16)
    at checkSourceElement (...\tsc.js:80140:7)
    ...
```

Related: #54442

**Comment by Andarist** [comment]
I'll investigate this and provide a fix/test case/whatever is needed here.

**Comment by fatcerberus** [comment]
@Andarist I guess Ryan was right in #54442 after all and the diagnostic really *was* reachable 😄

**Comment by Andarist** [comment]
yeah 😅 I made a wrong assumption that this holds true:
> so, in fact, we need a `baseClassType` to move forward to `resolveCall` and I think **that** this implies having at least one valid constructor which also implies that we won't pass an empty `signatures` list to `resolveCall`

What is interesting here is that this repro needs declaration merging **and** a class that `extends null` 🤯
